modules/lang-painless/src/main/java/org/elasticsearch/painless/ir/CallSubDefNode.java

```diff
@@ -45,50 +44,49 @@ public String getName() {
         return name;
     }
 
-    public void setRecipe(String recipe) {
-        this.recipe = recipe;
-    }
+    /* ---- end node data ---- */
 
-    public String getRecipe() {
-        return recipe;
-    }
+    @Override
+    protected void write(ClassWriter classWriter, MethodWriter methodWriter, ScopeTable scopeTable) {
+        methodWriter.writeDebugInfo(location);
 
-    public void addPointer(String pointer) {
-        pointers.add(pointer);
-    }
+        StringBuilder defCallRecipe = new StringBuilder();
+        List<Object> boostrapArguments = new ArrayList<>();
+        List<Class<?>> typeParameters = new ArrayList<>();
+        int capturedCount = 0;
 
-    public List<String> getPointers() {
-        return pointers;
-    }
+        typeParameters.add(Object.class);
 
-    public void addTypeParameter(Class<?> typeParameter) {
-        typeParameters.add(typeParameter);
-    }
+        for (int i = 0; i < getArgumentNodes().size(); ++i) {
+            ExpressionNode argumentNode = getArgumentNodes().get(i);
+            argumentNode.write(classWriter, methodWriter, scopeTable);
 
-    public List<Class<?>> getTypeParameters() {
-        return typeParameters;
-    }
+            typeParameters.add(argumentNode.getExpressionType());
 
-    /* ---- end node data ---- */
+            if (argumentNode instanceof DefInterfaceReferenceNode) {
```

Comment:
This is an unknown method ref that we have to resolve at runtime.

Reply:
commented